app/controllers/publishers/ats_api/v1/vacancies_controller.rb

```diff
@@ -1,58 +1,83 @@
 class Publishers::AtsApi::V1::VacanciesController < Api::ApplicationController
   before_action :authenticate_client!
 
+  rescue_from StandardError, with: :render_server_error
+  rescue_from ActiveRecord::RecordNotFound, with: :render_not_found
+  rescue_from ActionController::ParameterMissing, with: :render_bad_request
+
   def index
-    @pagy, @vacancies = pagy(vacancies, items: 100)
+    @pagy, @vacancies = pagy(vacancies.where(publisher_ats_api_client: client), items: 100)
 
-    respond_to(&:json)
+    render json: {
+      data: @vacancies.map { |vacancy| render_vacancy(vacancy) },
+      meta: { totalPages: @pagy.pages, count: @pagy.page },
+    }
   end
 
   def show
-    @vacancy = vacancy
+    vacancy = Vacancy.find(params[:id])
+    render json: render_vacancy(vacancy)
   end
 
-  # No idea why rubocop can't see the check after the create call
-  # rubocop:disable Rails/SaveBang
   def create
-    @vacancy = Vacancy.create(vacancy_params)
-
-    respond_to do |format|
-      format.json do
-        if @vacancy.persisted?
-          render status: :created
-        else
-          render status: :bad_request
-        end
-      end
+    result = Publishers::AtsApi::V1::CreateVacancyService.new(permitted_vacancy_params).call
+
+    if result[:headers]
```

Comment:
is this idiomatic to alter the headers hash in this way? Could we have not just put an error in the JSON instead? It might be hard to document this behaviour, so clients might miss it?

Reply:
https://stackoverflow.com/questions/9742380/link-headers-vs-link-elements-for-restful-json suggests that 'link' headers are not widely adopted - I hadn't heard of them until now. I'm just not sure how we would document the existence of such a header (compared to putting it in the JSON body) so that the client knows to look for it